common/containers/Tabs/Contracts/components/Interact/components/InteractForm/index.tsx

```diff
@@ -76,6 +80,51 @@ class InteractForm extends Component<Props, State> {
     return contracts && contracts.length;
   };
 
+  public componentDidMount = () => {
+    this.timer = setTimeout(
+      function() {
+        this.contractSetup();
+      }.bind(this),
+      1000
+    );
+  };
+
+  public componentWillUnmount = () => {
+    if (this.timer) {
+      clearTimeout(this.timer);
+    }
+  };
+
+  public getParamsFromUrl = () => {
```

Comment:
This function seems to be repeated 3x. Is there any way that you can extract this out, or use it in a higher order component and pass the data down as props?

Reply:
Took me longer than i want to admit but I managed to extract it! Tell me what you think about it!